homeassistant/components/stream/__init__.py

```diff
@@ -394,25 +398,24 @@ def _run_worker(self) -> None:
                 redact_credentials(str(self.source)),
             )
 
-        @callback
-        def worker_finished() -> None:
+        async def worker_finished() -> None:
             # The worker is no checking availability of the stream and can no longer track
             # availability so mark it as available, otherwise the frontend may not be able to
             # interact with the stream.
             if not self.available:
                 self._async_update_state(True)
             for provider in self.outputs().values():
-                self.remove_provider(provider)
+                await self.remove_provider(provider)
```

Comment:
Can we gather this work?

Reply:
I don't think that it's necessary here since the only thing that is blocking is the worker close, and that will only happen on removing the last provider.

Reply:
Maybe add a comment why we don't gather the work?

Reply:
Yes, makes sense, will do that now